### DISCUSSION TITLE
MacOS Option+Left/Right escape codes look different than other terminals

### DISCUSSION THREAD

**Discussion by hood** [ghostty-org/ghostty#4194] [Feature Requests, Ideas] [answered]
Sorry in advance for the lack of approriate terminology, I lack expertise in this domain.

When using Ghostty I found out that my `Option+Left` and `Option+Right` combinations for skipping through whole words in my prompt were not working as intended. FYI I’m using `zsh` + `starship` + `tmux` in my setup, and the bindings are explicitly set in my custom configuration files like such:
```sh
bindkey '^[B' backward-word
bindkey '^[F' forward-word
```

After debubbing by launching `cat` and then pressing the combinations to see what codes would be output, using `tmux` to sync my `ghostty` instance with `kitty` and `iterm2` instances, I realised that the codes output when pressing the combinations in `ghostty` look different, as they use lowercase characters instead of uppercase ones.

For example, Option+Right on `iterm2` sends the following: `'^[F’`, while the same combination in `ghostty` sends `’^[f’`. So, to fix in my specific setup, all I had to do was:
```diff
bindkey '^[B' backward-word
bindkey '^[F' forward-word
+ bindkey ‘^[b' backward-word
+ bindkey ‘^[f' forward-word
```

Is this working as intended?

**Comment by ansidev** [comment, marked as answer]
My configurations

```
# ~/.zshrc
bindkey '^[B' backward-word
bindkey '^[F' forward-word
```

My outputs:

Apple Terminal (I make no custom configuration to this application)

```
$ cat
^[b^[f^C
```

Ghostty with default key binding: Same outputs as Apple Terminal.

The keybinding are the same between those applications.

FYI: Ghostty deffault keybinding for Alt+LeftArrow và Alt+RightArrow are:
```
# run ghostty +show-config --default | grep 'alt' to see them
keybind = alt+left=esc:b
keybind = alt+right=esc:f
```

I overwrote the default keybinding and got no issue.

```
keybind = alt+left=esc:B
keybind = alt+right=esc:F
```

Ghostty output:
```
$ cat
^[B^[F^C
```

I do not see any issue with those key binding configuration.
- ^[B vs ^[b
- ^[F vs ^[f


I guess the issue may come from your tmux configuration.
